File: src/ds-visuals/max-heap/application/heap-sort/code/heapsort.py
hp = __import__('heap');
#place heap.py (max_heap.py - name changed) in same directory

class HeapSort(object):
	def __init__(self, arr):
		super(HeapSort, self).__init__()
		self.arr = arr

	def printH(self):
		print(self.arr)

	def heapSort(self):
		heap = hp.Heap()
		heap.createHeap(*self.arr)
		i = 0
		while(heap.size > 0):
			self.arr[i] = heap.delete()
			i += 1

arr = [4, 13, 6, 2, 87, 21, 65]
heapSort = HeapSort(arr)
heapSort.printH()
heapSort.heapSort()
heapSort.printH()

File: src/ds-visuals/snippets/max-heap/heap_sort.c
7
very_fast =
fast =  0, 1, 3, 6
medium = 2, 4, 5, 7
slow = 
very_slow = 

int* heapSort(int* arr, int n){
	int i;
	Heap heap = createheap(n, arr);
	i = 0;
	while(heap.size > 0){
		arr[i] = delete(&heap);
		i+=1;
	}
	return arr;
}